### PR TITLE
[8] 2022/bar-chart

### PR DESCRIPTION
**Pull request by zastrow**
Creates the Bart Chart styles and adds them to the appropriate sections.

CleanShot 2022-06-13 at 15 33 22@2x

## What needs checked:
- [ ] Install and run the project using the README instructions
- [ ] Sass code looks good
- [ ] React code looks good
- [ ] Small screen / responsiveness of chart

### CODE REVIEW

**Review by reiniiriarios — COMMENTED** [review]
can't build on apple silicon 😮. can't find a workaround either, compiling fails.
```
npm ERR! ERR! sharp Prebuilt libvips 8.10.5 binaries are not yet available for darwin-arm64v8
```

I left one comment nonetheless.
From the screenshots and deploy preview, this is looking great! 🏆

File: src/components/2022/bar-chart.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    dataBars.push(
+      <div className="cmp-bar-chart" key={i} style={cssProperties}>
+        <div className="cmp-bar-chart__title">
+          {title}
+        </div>
+        <div className="cmp-bar-chart__value">
+          {value}%
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (zastrow) [reply src/components/2022/bar-chart.js:24]: I 100% agree that is what we would want to do if this needed to be more modular. Nice thing with this site, it’ single use. So we can be prescriptive in this case.

**Comment by zastrow** [comment]
@reiniiriarios Thank you for reviewing this! With the M1 Mac issues, check out the project README, there is a note about troubleshooting for an M1 Mac under the Setup.

**Comment by reiniiriarios** [comment]
@zastrow Ack! I don't know how my brain skipped right over that, thanks. This has sent me down an interesting rabbit hole, however.

Next problem was that sharp, a dependency of gatsby, needed python 2 to install. The following solved that issue, if you think it should be jotted down anywhere:
```
brew install pyenv
pyenv install 2.7.18
pyenv global 2.7.18
export PYTHON=$(pyenv root)/shims/python
```

But now I'm getting c++ compile errors when `make` fails after `npm i` calls `npm-gyp` to to compile `node-sass`. It looks like `node-sass` doesn't support M1 chips yet either.

There's probably a workaround, but you seem to have plenty of other people jumping in without having issues, so I might leave it in their hands. I thought I'd update you regardless!

**Comment by reiniiriarios** [comment]
@bengoshow Reinstalling `vips` did solve the `libvps` error, but it failed again later. Brew will install python 3, but sharp needed 2 (it was failing on a `print` syntax error). After solving that issue, I'm at the following.

```
npm ERR! make: *** [Release/obj.target/binding/src/binding.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/Users/emmalitwavulcu/Code/designsystemssurvey/node_modules/node-gyp/lib/build.js:262:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Darwin 21.5.0
npm ERR! gyp ERR! command "/opt/homebrew/Cellar/node/18.3.0/bin/node" "/Users/emmalitwavulcu/Code/designsystemssurvey/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /Users/emmalitwavulcu/Code/designsystemssurvey/node_modules/node-sass
npm ERR! gyp ERR! node -v v18.3.0
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok 
npm ERR! Build failed with error code: 1
```

Which led me here: https://github.com/sass/node-sass/issues/3033

Since `node-sass` is a dependency somewhere down the chain, I'm not sure how to work around it (such as replacing it with `sass`). The `gatsby` version in `package.json` is 2.*, and 4 is released--I wonder if an update might solve the issue (and/or cause more).

**Comment by zastrow** [comment]
Thank you for your help on this PR, we’re moving forward with final reviews on the main feature branch going into main with #354.

**Pull request closed without merging**